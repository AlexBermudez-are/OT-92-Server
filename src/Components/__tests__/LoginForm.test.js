import LoginForm from "../Auth/LoginForm";
import {fireEvent, 
  render, 
  screen,
} from "@testing-library/react";
import renderer from "react-test-renderer";

test("Should render emailError", () => {
  render(<LoginForm />);
  const inputEmail = screen.getByTestId("emailError");
  fireEvent.change(inputEmail, {target: {value: ""}});
  expect(inputEmail.value).toBe("");
});

test("Should render passwordError", () => {
  render(<LoginForm />);
  const inputPassword = screen.getByTestId("passwordError");
  fireEvent.change(inputPassword, {target: {value: ""}});
  expect(inputPassword.value).toBe("");
});

test("matches snapshot", () => {
  const tree = renderer.create(<LoginForm />).toJSON();
  expect(tree).toMatchSnapshot();
});