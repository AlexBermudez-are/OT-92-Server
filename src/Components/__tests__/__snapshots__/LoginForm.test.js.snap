// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="form-group"
>
  <h1
    className="h1 text-center mt-5"
  >
     Ingresar 
  </h1>
  <form
    className="form-container"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div>
      <label
        className="form-label"
        htmlFor="email "
      >
        Email
      </label>
      <input
        className="w-100 form-control"
        data-testid="emailError"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Ingrese email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="mt-4"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Contraseña
      </label>
      <input
        className="w-100 form-control"
        data-testid="passwordError"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Ingrese contraseña"
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary mt-5 m-auto"
      disabled={false}
      style={
        Object {
          "width": "10rem",
        }
      }
      type="submit"
    >
      Ingresar
    </button>
  </form>
</div>
`;
